Error when switching between topics

When switching between image topics (in my case especially when switching from a topic with rgb images to a topic with depth images) sometimes a segmentation fault occurs. I think [this is the problematic line](https://github.com/ros-visualization/rqt_image_view/blob/fe076acd265a05c11c04f9d04392fda951878f54/src/rqt_image_view/image_view.cpp#L598) and the error occurs when something goes out of scope [here](https://github.com/ros-visualization/rqt_image_view/blob/fe076acd265a05c11c04f9d04392fda951878f54/src/rqt_image_view/image_view.cpp#L609). Maybe a destructor is called on something that should not be destructed?

I am not sure what the problem is. After further testing, I get a variety of errors, e.g:
`OpenCV Error: Assertion failed (tlsSlotsSize > slotIdx) in getData, file /tmp/binarydeb/ros-kinetic-opencv3-3.3.1/modules/core/src/system.cpp, line 1425
terminate called after throwing an instance of 'cv::Exception'
terminate called recursively
  what():  /tmp/binarydeb/ros-kinetic-opencv3-3.3.1/modules/core/src/system.cpp:1425: error: (-215) tlsSlotsSize > slotIdx in function getData`
or
`Error in python: corrupted double-linked list`
or
`Error in python: free(): invalid pointer`
My best guess at the moment is some kind of race condition between threads and the callback function being called too frequently.

--- a/src/rqt_image_view/image_view.cpp
+++ b/src/rqt_image_view/image_view.cpp
@@ -321,6 +321,8 @@ void ImageView::selectTopic(const QString& topic)
 
 void ImageView::onTopicChanged(int index)
 {
+  conversion_mat_.release();
+
   subscriber_.shutdown();
 
   // reset image on topic change

